### PR TITLE
TestCase Theory Example Reduce code duplication.. make it dry

### PR DESCRIPTION
**Pull request by mkbmain**
TestCase Theory Example Reduce code duplication

### CODE REVIEW

File: Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs [hunk 1]
```diff
@@ -24,57 +24,20 @@ public TestHireCommandHandler()
...
-    [Fact]
-    public void Test_NoParameters()
+    [Theory]
+    [InlineData("NoParameters", "")]
+    [InlineData("OneNumericParameter", "1")]
+    [InlineData("OneInvalidTextParameter", "asd")]
+    [InlineData("TwoTextParameters", "asd qwe")]
+    [InlineData("TwoValidButNoQty", "Sales Production")]
+    [InlineData("ThreeParameters", "1 2 3")]
+    [InlineData("InvalidJobType", "CEO 1")]
```
Comment (mkbmain) [inline Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs:34]: name just used for readability with in test runner 
i.e if a case fails what case where we meant to be testing
image

Comment (mkbmain) [reply Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs:34]: I can remove it if your not fussed

File: Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs [hunk 2]
```diff
@@ -24,57 +24,20 @@ public TestHireCommandHandler()
...
-        Assert.Equal(DEFAULT_CHATTER_DISPLAY_NAME, 
-            chatMessageEvent.Arguments.ChatterDisplayName);
-        Assert.Equal(Literals.YouDoNotHaveACompany, 
-            chatMessageEvent.Arguments.Message);
-    }
-
-    [Fact]
-    public void Test_NoParameters()
+    [Theory]
+    [InlineData("NoParameters", "")]
```
Comment (ScottLilly) [inline Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs:28]: The "name" parameter is not being used anywhere that I see. Is it needed?

Comment (mkbmain) [reply Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs:28]: I have a comment about this with screen shot just above ... its not needed but adds clarity to what you are testing with in the test runner. I am fine to remove it .. BUT I find a lot of time with tests the hard part is not making them green its answering WHY they are there in the first place a name helps that.

i.e xyc fails buy xyz passes what where we testing with a name that is communicated better imo.

Comment (mkbmain) [reply Test.MegaCorpClash.Models/ChatCommandHandlers/TestHireCommandHandler.cs:28]: if this answers your question feel free to mark this as resolved if not hola at me il remove the name from params

**Review by ScottLilly — APPROVED** [review]

**Review by ScottLilly — APPROVED** [review]

**Pull request merged**